### PR TITLE
Add Rails quick start

### PR DESCRIPTION
**Pull request by CaroFG**
# Pull Request

## Related issue
-
## What does this PR do?
- Adds Rails quick start + assets
- Creates the dedicated sidebar route in the backend section of guides

## PR checklist
Please check if your PR fulfills the following requirements:
- [x] Does this PR fix an existing issue, or have you listed the changes applied in the PR description (and why they are needed)?
- [x] Have you read the contributing guidelines?
- [x] Have you made sure that the title is accurate and descriptive of the changes?

Thank you so much for contributing to Meilisearch!

### CODE REVIEW

File: guides/ruby_on_rails_quick_start.mdx [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+description: Integrate Meilisearch into your Ruby on Rails app.
+---
+
+# Ruby on Rails quick start
+
+Integrate Meilisearch into your Ruby on Rails app.
+
+## 1. Create a Meilisearch project
+
+[Create a project](https://cloud.meilisearch.com) in the Meilisearch Dashboard. Check out our [getting started guide](https://www.meilisearch.com/docs/learn/getting_started/cloud_quick_start) for step-by-step instructions.
```
Comment (Strift) [inline guides/ruby_on_rails_quick_start.mdx:12]:
```suggestion
[Create a project](https://cloud.meilisearch.com) in the Meilisearch Dashboard. Check out our [getting started guide](/learn/getting_started/cloud_quick_start) for step-by-step instructions.
```

File: guides/ruby_on_rails_quick_start.mdx [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+
+## 4. Add your Meilisearch credentials
+
+Run the following command to create a `config/initializers/meilisearch.rb` file.
+
+```bash
+bin/rails meilisearch:install
+```
+
+Then add your Meilisearch URL and [Default Admin API Key](https://www.meilisearch.com/docs/learn/security/basic_security#obtaining-api-keys). On Meilisearch Cloud, you can find your credentials in your project settings.
```
Comment (Strift) [inline guides/ruby_on_rails_quick_start.mdx:38]:
```suggestion
Then add your Meilisearch URL and [Default Admin API Key](/learn/security/basic_security#obtaining-api-keys). On Meilisearch Cloud, you can find your credentials in your project settings.
```

File: guides/ruby_on_rails_quick_start.mdx [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+```Ruby
+# Meilisearch is typo-tolerant:
+hits = Article.search('deepre')
+hits.first
+```
+We strongly recommend using the frontend search to enjoy the swift and responsive search-as-you-type experience.
+
+### Frontend search
+
+For testing purposes, you can explore the records using our built-in [search preview](https://www.meilisearch.com/docs/learn/getting_started/search_preview).
```
Comment (Strift) [inline guides/ruby_on_rails_quick_start.mdx:117]:
```suggestion
For testing purposes, you can explore the records using our built-in [search preview](/learn/getting_started/search_preview).
```

File: guides/ruby_on_rails_quick_start.mdx [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+
+![Searching through Rails table data with Meilisearch search preview UI](/assets/images/rails_quick_start/rails-qs-search-preview.gif)
+
+We also provide resources to help you quickly build your own [frontend interface](https://www.meilisearch.com/docs/guides/front_end/front_end_integration).
```
Comment (Strift) [inline guides/ruby_on_rails_quick_start.mdx:121]:
```suggestion
We also provide resources to help you quickly build your own [frontend interface](/guides/front_end/front_end_integration).
```

File: guides/ruby_on_rails_quick_start.mdx [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+
+## Next steps
+
+Configure your [index settings](https://www.meilisearch.com/docs/learn/configuration/settings). For a full configuration example, see the [meilisearch-rails gem README](https://github.com/meilisearch/meilisearch-rails?tab=readme-ov-file#%EF%B8%8F-settings).
```
Comment (Strift) [inline guides/ruby_on_rails_quick_start.mdx:125]:
```suggestion
Configure your [index settings](/learn/configuration/settings). For a full configuration example, see the [meilisearch-rails gem README](https://github.com/meilisearch/meilisearch-rails?tab=readme-ov-file#%EF%B8%8F-settings).
```

**Review by Strift — COMMENTED** [review]
LGTM. I suggest converting the links to docs to internal MD links though. It's slightly better for authors and I want to make things as easy as possible for the linter lol

**Comment by CaroFG** [comment]
> LGTM. I suggest converting the links to docs to internal MD links though. It's slightly better for authors and I want to make things as easy as possible for the linter lol

I completely forgot about that, sorry!

**Review by Strift — APPROVED** [review]

File: guides/ruby_on_rails_quick_start.mdx [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+description: Integrate Meilisearch into your Ruby on Rails app.
+---
+
+# Ruby on Rails quick start
+
+Integrate Meilisearch into your Ruby on Rails app.
+
+## 1. Create a Meilisearch project
+
+[Create a project](https://cloud.meilisearch.com) in the Meilisearch Dashboard. Check out our [getting started guide](/learn/getting_started/cloud_quick_start) for step-by-step instructions.
```
Comment (guimachiavelli) [inline guides/ruby_on_rails_quick_start.mdx:12]:
```suggestion
[Create a project](https://cloud.meilisearch.com) in the Meilisearch Cloud dashboard. Check out our [getting started guide](/learn/getting_started/cloud_quick_start) for step-by-step instructions.
```

File: guides/ruby_on_rails_quick_start.mdx [hunk 7]
```diff
@@ -0,0 +1,125 @@
+---
+title: Ruby on Rails quick start — Meilisearch documentation
+description: Integrate Meilisearch into your Ruby on Rails app.
+---
+
+# Ruby on Rails quick start
```
Comment (guimachiavelli) [inline guides/ruby_on_rails_quick_start.mdx:6]: I recommend making it clear from the start this guide is focussed on Meilisearch Cloud. Don't need to make a big fuss out of it, just mention it early on.

File: guides/ruby_on_rails_quick_start.mdx [hunk 8]
```diff
@@ -0,0 +1,125 @@
...
+```Ruby
+MeiliSearch::Rails.configuration = {
+  meilisearch_url: '<your Meilisearch URL>',
+  meilisearch_api_key: '<your Meilisearch API key>'
+}
+```
+
+## 5. Generate the model and run the database migration
+
+Create an example Article model and generate the migration files.
```
Comment (guimachiavelli) [inline guides/ruby_on_rails_quick_start.mdx:49]: Seems like we use backticks when mentioning the model elsewhere in this page, so better to be consistent.

```suggestion
Create an example `Article` model and generate the migration files.
```

File: guides/ruby_on_rails_quick_start.mdx [hunk 9]
```diff
@@ -0,0 +1,125 @@
...
+
+For testing purposes, you can explore the records using our built-in [search preview](/learn/getting_started/search_preview).
+
+![Searching through Rails table data with Meilisearch search preview UI](/assets/images/rails_quick_start/rails-qs-search-preview.gif)
+
+We also provide resources to help you quickly build your own [frontend interface](/guides/front_end/front_end_integration).
+
+## Next steps
+
+Configure your [index settings](/learn/configuration/settings). For a full configuration example, see the [meilisearch-rails gem README](https://github.com/meilisearch/meilisearch-rails?tab=readme-ov-file#%EF%B8%8F-settings).
```
Comment (guimachiavelli) [inline guides/ruby_on_rails_quick_start.mdx:125]: Isn't it better to configure index settings before adding documents so Meilisearch doesn't have to re-index everything?

Perhaps you can replace "Next steps" with a "Conclusion" recapping all the steps?

**Review by guimachiavelli — DISMISSED** [review]
I'm preemptively approving this, but I did leave a few comments that might require an extra commit or two (if you decide they're relevant, of course).

File: guides/ruby_on_rails_quick_start.mdx (hunk 7 above)
Comment (CaroFG) [reply guides/ruby_on_rails_quick_start.mdx:6]: It also works with self-hosted Meilisearch. What do you think of adding the following sentence in step 1?

`If you prefer to use the self-hosted version of Meilisearch, you can follow the[ quick start](https://www.meilisearch.com/docs/learn/getting_started/quick_start) tutorial.`

**Review by Strift — APPROVED** [review]

File: guides/ruby_on_rails_quick_start.mdx (hunk 7 above)
Comment (guimachiavelli) [reply guides/ruby_on_rails_quick_start.mdx:6]: Sounds good to me!

**Review by guimachiavelli — APPROVED** [review]
LGTM!

**Pull request merged**